### PR TITLE
chore(examples): adding readmes to custom parser and file header examples

### PR DESCRIPTION
**Pull request by dbanksdesign**
*Issue #, if available:* #596

*Description of changes:* The custom parser and custom file header examples did not have README files. This fixes that. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/advanced/custom-file-header/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Custom File Header
+
+Style Dictionary 3.0 added the ability to define custom file headers for files generated from formats. A file header is a comment at the beginning of the generated file that has some information about how it was generated. This is fairly common in build tools that output source code. Before 3.0 you could only show or hide this header in the built-in formats. Now you can write your own custom file header messages that can be used in built-in formats as well as custom formats. 
+
+Common use cases include:
+
+- Using a version number in the file header comment so that you can check in files generated by Style Dictionary to git
+- Using a hash of the source in the file header comment so that it only changes when there is a change in the source
```
Comment (chazzmoney) [inline examples/advanced/custom-file-header/README.md:8]: I think this is what you meant?

> it only changes

it always changes

File: examples/advanced/custom-file-header/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+1. Using the `registerFileHeader` method, and then referencing the name in the file configuration.
+1. Adding a `fileHeader` object to the Style Dictionary configuration, and then referencing the key in the file configuration
+1. Writing the `fileHeader` as a function in the platform or file configuration
+
+This example codebase has all 3 methods for you to see which works best for you. All 3 methods are in the [**build.js**](/build.js) file. 
+
+A file header is a function that returns an array of strings. This array of strings will get mapped to lines in a comment at the beginning of a file generated by a formatter. The formatter will take care of how to format the lines in the proper comment style (`//`, `/* */`).
+
+You can reference the file header in a custom format as well by using the `fileHeader` function in `StyleDictionary.formatHelpers`.
```
Comment (chazzmoney) [inline examples/advanced/custom-file-header/README.md:31]: , as shown below:

File: examples/advanced/custom-parser/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+## Custom Parsers
+
+Style Dictionary 3.0 added the ability to define custom parsers for source files. Before 3.0, the source files for Style Dictionary had to be JSON, JSON5, or Node modules (unless you wanted to read and merge the data yourself outside of Style Dictionary). Now you can define custom parsers that run on certain files based on a file path pattern regular expression (similar to how Webpack loaders work). 
+
+Common use cases include:
```
Comment (chazzmoney) [inline examples/advanced/custom-parser/README.md:5]: Example use cases include:

File: examples/advanced/custom-file-header/README.md (hunk 1 above)
Comment (dbanksdesign) [reply examples/advanced/custom-file-header/README.md]: Hopefully this sounds better?

**Comment by chazzmoney** [comment]
LGTM! :shipit:

**Pull request merged**